crates/dservice/src/manager.rs

```diff
@@ -284,17 +289,22 @@ impl Manager {
                         );
                     }
                     account.remaining_task -= 1;
-                    if account.remaining_task == 0 {
-                        should_clear_account = true;
+                    if account.remaining_task % 5000 == 0 {
+                        update_account = true;
+                    }
+                    if task_info.sequence > latest_scanned_block.sequence as i64 {
+                        latest_scanned_block = BlockInfo {
+                            hash: task_info.hash.clone(),
+                            sequence: task_info.sequence as u64,
```

Comment:
`BlockInfo` should be refactored in subsequent PR to be a `i64` for sequence, as negative block sequences are not possible.